src/execute_request.jl

```diff
@@ -5,54 +5,111 @@
 import Base.Libc: flush_cstdio
 import Pkg
 
-const text_plain = MIME("text/plain")
-const image_svg = MIME("image/svg+xml")
-const image_png = MIME("image/png")
-const image_jpeg = MIME("image/jpeg")
-const text_markdown = MIME("text/markdown")
-const text_html = MIME("text/html")
-const text_latex = MIME("text/latex") # Jupyter expects this
-const text_latex2 = MIME("application/x-latex") # but this is more standard?
-const application_vnd_vega_v3 = MIME("application/vnd.vega.v3+json")
-const application_vnd_vegalite_v2 = MIME("application/vnd.vegalite.v2+json")
-const application_vnd_dataresource = MIME("application/vnd.dataresource+json")
+Base.showable(a::Vector{MIME}, x) = any(m -> showable(m, x), a)
+
+"""
+A vector of MIME types (or vectors of MIME types) that IJulia will try to
+render. IJulia will try to render every MIME type specified in the first level
+of the vector. If a vector of MIME types is specified, IJulia will include only
+the first MIME type that is renderable (this allows for the expression of
+priority and exclusion of redundant data).
+
+For example, since "text/plain" is specified as a first-child of the array,
+IJulia will always try to include a "text/plain" representation of anything that
+is displayed. Since markdown and latex are specified within a sub-vector, IJulia
+will always try to render "text/markdown", and will only try to render
+"text/latex" if markdown isn't possible.
+"""
+const ijulia_mime_types = Vector{Union{MIME, Vector{MIME}}}([
+    MIME("text/plain"),
+    MIME("image/svg+xml"),
+    MIME("image/png"),
+    MIME("image/jpeg"),
+    MIME("text/html"),
+    [
+        MIME("text/markdown"),
+        MIME("text/latex"), # Jupyter expects this
+        MIME("application/x-latex"), # but this is more standard?
+    ],
+])
+
+"""
+MIME types that when rendered (via stringmime) return JSON data. See
+`ijulia_mime_types` for a description of how MIME types are selected.
+
+This is necessary to embed the JSON as is in the displaydata bundle (rather than
+as stringify'd JSON).
+"""
+const ijulia_jsonmime_types = Vector{Union{MIME, Vector{MIME}}}([
+    [MIME("application/vnd.vegalite.v2+json"), MIME("application/vnd.vega.v3+json")],
+])
+
+register_ijulia_mime(x::Union{MIME, Vector{MIME}}) = push!(ijulia_mime_types, x)
```

Comment:
No _ijulia_ here since that is already in the module name.  i.e. `IJulia.register_mime`